ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/FeatureProjectFactory.java

```diff
@@ -440,7 +443,24 @@ public void onEnable(ProgressHandle progressHandle) {
             }
 
             public void onError(String message) {
-                error = message;
+                SpecificationVersion jdk = new SpecificationVersion(System.getProperty("java.specification.version"));
+                FeatureInfo[] fi = Arrays.copyOf(additional, additional.length + 1);
+                fi[additional.length] = info;
+                boolean required = false;
+
+                for (FeatureInfo i : fi) {
```

Comment:
This code snippet reminds me code snippet somewhere else. Consider extracting into a method.